app/services/HttpIdentityService.scala

```diff
@@ -95,9 +94,19 @@ case class GuestRegistrationResponse(
     guestRegistrationRequest: GuestRegistrationResponse.GuestRegistrationRequest
 )
 
+case class UserIdWithOptionalToken(userId: String, token: Option[String])
```

Comment:
Would be good to get a comment saying what the token is, or maybe a more specific name. And (out of interest) is it used to authenticate when creating a password for a guest account?